usaspending_api/search/tests/integration/spending_by_category/test_spending_by_recipient.py

```diff
@@ -0,0 +1,203 @@
+import logging
+import uuid
+
+from model_mommy import mommy
+from elasticsearch_dsl import A
+
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.search.tests.integration.spending_by_category.utilities import setup_elasticsearch_test
+
+
+def _make_fpds_transaction(id, award_id, obligation, action_date, recipient_duns, recipient_name):
+    # Transaction Normalized
+    mommy.make(
+        "awards.TransactionNormalized",
+        id=id,
+        is_fpds=True,
+        award_id=award_id,
+        federal_action_obligation=obligation,
+        action_date=action_date,
+    )
+
+    # Transaction FPDS
+    mommy.make(
+        "awards.TransactionFPDS",
+        transaction_id=id,
+        awardee_or_recipient_uniqu=recipient_duns,
+        awardee_or_recipient_legal=recipient_name,
+    )
```

Comment:
This looks like a resuable component that could be  added to a fixture  library somewhere, or taken from an existing one.

Reply:
Could be. In fact I stole it from [`usaspending_api.search.tests.integration.spending_by_category.spending_test_fixtures.awards_and_transactions`](https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/search/tests/integration/spending_by_category/spending_test_fixtures.py#L83) ... and repurposed it for this test. The "private" function there was just to cut down on extra line-wrapping when having this directly in the test.

I think it really should be more of the _pattern_ that people follow, and where a pattern is guaranteed to be reusable, we can externalize into a builder-utility. But the pattern (while a bit verbose) should be easy to copy-paste-adapt and isn't overly complex such that we'd want to centralize it in reusable code. In fact, centralizing it can sometimes cause trouble if someone was expecting their test to setup data in a subtle way (which I am).

Reply:
While you are mocking a different set of variables than that other fixture, the relationship between transaction normalized and transaction FPDS tables certainly isn't peculiar to this test. Both this fixture (and the fixture you took it from) are  tech debt that we'll need to spend time paying off if we ever get around to refactoring the transaction tables.

Reply:
@BrianZito they're not debt until we refactor the tables and _don't_ fix the tests. Then they're debt we chose to incur. 

At this point they are a potential liability. My point above is that I think there are many tradeoffs in liabilities, and this is a lesser liability than reusing code that may undergo changes/additions by others using and expanding the same code, which then may break these tests. So I made data handling a private concern for this test, not a shared concern -- similar to OOP encapsulation.